docs/apis/subsystems/external/writing-a-service.md

```diff
@@ -332,10 +332,10 @@ public static function get_biscuit_parameters() {
 public static function get_biscuit_parameters() {
     return new external_function_parameters([
         'ifeellike' => new external_single_structure([
-                'chocolatechips' => new external_value(PARAM_BOOL, VALUE_REQUIRED),
-                'glutenfree' => new external_value(PARAM_BOOL, PARAM_DEFAULT, false),
+                'chocolatechips' => new external_value(PARAM_BOOL, 'if biscuit contains chocolate chips', VALUE_REQUIRED),
```

Comment:
VALUE_REQUIRED also wrong.

Reply:
why is this wrong?

Reply:
Apologies- I think this one is right. Just would benefit from switching to named args in php.

Reply:
so you are suggesting to switch to named args in the whole example? I just did it for 'glutenfree' to demonstrate that both is possible.